Tests/AsyncQueueTests/FIFOQueueTests.swift

```diff
@@ -46,6 +46,32 @@ final class FIFOQueueTests: XCTestCase {
         await systemUnderTest.await { /* Drain the queue */ }
     }
 
+    func test_asyncOn_sendsEventsInOrder() async {
```

Comment:
Most of these new tests mirror the tests directly above them that tested the API without the `on isolatedActor` parameter.